internal/pkg/source/source.go

```diff
@@ -109,13 +109,25 @@ func (a *Source) visitOperands(operands []*ssa.Value) {
 		if !ok || a.marked[n] {
 			continue
 		}
-		al, ok := (*o).(*ssa.Alloc)
-		if !ok || al.Comment == "slicelit" || a.config.IsSource(al.Type()) {
-			a.dfs(n)
+		al, isAlloc := (*o).(*ssa.Alloc)
+		// An Alloc represents the allocation of space for a variable. If a Node is an Alloc,
+		// and the thing being allocated is not a slice, then either:
+		// a) it is a Source value, in which case it will get its own DFS
+		// b) it is not a Source value, in which case we should not DFS through it.
+		// However, if the Alloc is a slice, then that means the source that we are DFSing from
```

Comment:
Could we be certain that the `Alloc` is for a `Source`?

Reply:
I'm not sure I understand your question.

We could check its type, yes. But if it is a `Source` it will get its own DFS. If it isn't a `Source`, then we shouldn't DFS through it, because that would mean we are tainting a new variable that isn't a `Source`.